BrainPortal/public/javascripts/userfiles.js

```diff
@@ -722,25 +722,45 @@ $(function() {
         }
       }
 
-      if (window.FileReader && $('#upload-dialog').data('max-upload-size'))
-        /* check selected files against upload limit */
+      if (window.FileReader)
+        // Checks the file for problems, and disables the upload button if:
+        //   - no file selected
+        //   - spaces in filename
+        //   - filesize greater than maximum allowed by server
         $('#upload-dialog')
-          .undelegate('#up-file', 'change.uf.up-limit')
-          .delegate(  '#up-file', 'change.uf.up-limit', function () {
+          .undelegate('#up-file', 'change.uf.file-problem')
+          .delegate(  '#up-file', 'change.uf.file-problem', function () {
+            var warning_text = "";
             var max = parseInt($('#upload-dialog').data('max-upload-size'));
+            var bad_file;
+            var selected = !!$(this).val();
 
-            var visible = (
-              this.files && this.files[0] &&
-              max && this.files[0].size > max
-            );
+            var spaces_in_filename = $(this).val().includes(" ");
+            var file_too_big;
+            if ( max > 0 ){
+              file_too_big = this.files && this.files[0] && max && this.files[0].size > max
+            } else {
+              file_too_big = false;
+            }
+
+            bad_file = ( spaces_in_filename || file_too_big );
+
+            if ( spaces_in_filename ) {
+              warning_text += "Spaces in filename not allowed! ";
```

Comment:
I have tried to use the userfiles model check for file validity, but I don't understand this RegEx:
`  def self.is_legal_filename?(basename)
    return true if basename && basename.match(/^[a-zA-Z0-9][\w\~\!\@\#\%\^\&\*\(\)\-\+\=\:\[\]\{\}\|\<\>\,\.\?]*$/)
    false
  end`

It appears to allow spaces?